### PR TITLE
Use changesets for releases

### PR DESCRIPTION
**Pull request by rygine**
in this PR:

- added changesets tooling for package releases
- removed `semantic-release` dependency
- updated some package names
- removed no longer needed files
- added `.node-version` file and updated node version
- added `LICENSE` files
- added code of conduct and contribution guide
- updated GitHub release workflow

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -6,32 +6,35 @@ on:
       - beta
 jobs:
   release:
+    permissions:
```
Comment (neekolas) [inline .github/workflows/release.yml:9]: You'll need the `id-token` for NPM provenance, which I'd like to make a standard practice across XMTP

File: package.json [hunk 2]
```diff
@@ -1,30 +1,24 @@
 {
-  "name": "@metamask/template-snap-monorepo",
-  "version": "0.2.1",
+  "name": "@xmtp/snap-monorepo",
```
Comment (neekolas) [inline package.json:2]: Thanks for removing this generated boilerplate

**Review by neekolas — APPROVED** [review]
Approving, but I believe the NPM publishing will fail without the `id-token` because provenance is set to true in the Snap package.json.

File: .github/workflows/release.yml (hunk 1 above)
Comment (rygine) [reply .github/workflows/release.yml:9]: good catch! i added it back.

**Pull request merged**